crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -322,7 +322,66 @@ impl TransactionResponse for Transaction {
     fn input(&self) -> &Bytes {
         &self.input
     }
+
+    fn gas_price(tx: &impl alloy_consensus::Transaction, base_fee: Option<u64>) -> u128 {
+        match TxType::try_from(tx.ty()).expect("should decode") {
+            TxType::Legacy | TxType::Eip2930 => tx.max_fee_per_gas(),
+            TxType::Eip1559 | TxType::Eip4844 | TxType::Eip7702 => {
+                // the gas price field for EIP1559 is set to `min(tip, gasFeeCap - baseFee) +
+                // baseFee`
+                base_fee
+                    .and_then(|base_fee| {
+                        tx.effective_tip_per_gas(base_fee).map(|tip| tip + base_fee as u128)
+                    })
+                    .unwrap_or_else(|| tx.max_fee_per_gas())
+            }
+        }
+    }
+
+    fn max_fee_per_gas(tx: impl alloy_consensus::Transaction) -> Option<u128> {
+        match TxType::try_from(tx.ty()).expect("should decode") {
+            TxType::Legacy | TxType::Eip2930 => None,
+            TxType::Eip1559 | TxType::Eip4844 | TxType::Eip7702 => Some(tx.max_fee_per_gas()),
+        }
+    }
```

Comment:
this implementation varies from the impl on `alloy_consensus::Transaction`. this code is ported from reth, without modifying the logic @mattsse

Reply:
https://github.com/paradigmxyz/reth/blob/e6dc947a3728637aa3b0f78c893e8f24dc468357/crates/rpc/rpc-types-compat/src/transaction/mod.rs#L41-L60